### PR TITLE
🔥 remove new_token_ids from warmup decode

### PR DESCRIPTION
**Pull request by prashantgupta24**
# Description

Fix the compiler issue with the new changes

### CODE REVIEW

File: vllm_spyre/v1/worker/spyre_model_runner.py [hunk 1]
```diff
@@ -879,7 +879,9 @@ def _prepare_decode(
             slot = [start_slot + offset]
             slot_mapping.append(slot)
             output_token_ids = req_state.output_token_ids
-            generation_token = output_token_ids[-1]
+            generation_token = torch.tensor(
+                output_token_ids[-1], dtype=torch.long, device=self.device
+            )
```
Comment (prashantgupta24) [inline vllm_spyre/v1/worker/spyre_model_runner.py:884]: This was a difference with the earlier code, but still not working 🤔

**Comment by prashantgupta24** [comment]
Also one concern is how these bugs slipped past the cpu tests

File: vllm_spyre/v1/worker/spyre_model_runner.py [hunk 2]
```diff
@@ -879,7 +879,9 @@ def _prepare_decode(
             slot = [start_slot + offset]
             slot_mapping.append(slot)
             output_token_ids = req_state.output_token_ids
-            generation_token = output_token_ids[-1]
+            generation_token = torch.tensor(output_token_ids[-1],
+                                            dtype=torch.long,
+                                            device=self.device)
```
Comment (yannicks1) [inline vllm_spyre/v1/worker/spyre_model_runner.py:884]: I think we should revert this: `input_tokens` is a list and gets converted to a tensor only in line 890/892... 
Also, lines 881/882/883 can be consolidated into one line.

File: vllm_spyre/v1/worker/spyre_worker.py [hunk 3]
```diff
@@ -359,21 +359,15 @@ def _warmup_spyre_dynamic_size(self, special_token_ids):
...
-                    valid_token_ids_tensor[torch.randint(
-                        0, len(valid_token_ids_tensor), (1, )).item()]
-                ])  # placeholder token
-                new_block_ids.append([req.block_ids])
                 num_computed_tokens.append(prompt_len)
             cached_request_data = CachedRequestData(
                 req_ids=req_ids,
                 resumed_from_preemption=False,
-                new_token_ids=new_token_ids,
+                new_token_ids=[[] for _ in range(len(dummy_requests))],
```
Comment (joerunde) [inline vllm_spyre/v1/worker/spyre_worker.py:370]: does this need to be set to anything? I'm actually not clear on whether or not the prefill pass returns a first sampled token which may be cached here

Comment (prashantgupta24) [reply vllm_spyre/v1/worker/spyre_worker.py:370]: The prefill pass does return a sampled token, but that caching happens within the execute_model And hence we don't need to pass in `new_token_ids` anymore.

**Comment by prashantgupta24** [comment]
No hurry as such, don't want to add anything before the release lol

**Comment by prashantgupta24** [comment]
bot:test
MARKERS="spyre and not quantized"

**Review by maxdebayser — APPROVED** [review]
LGTM

**Comment by prashantgupta24** [comment]
bot:test
MARKERS="spyre and cb"

**Comment by prashantgupta24** [comment]
bot:test
MARKERS="spyre and cb"

**Comment by prashantgupta24** [comment]
Failed 2 different tests this time - 

```tests/e2e/test_spyre_cb_scheduler_steps.py::test_prompt_too_long_for_current_tkv[sendnn-ibm-ai-platform/micro-g3.3-8b-instruct-1b] FAILED```
```tests/e2e/test_spyre_cb_scheduler_steps.py::test_requested_tokens_not_fitting_remaining_space[sendnn-ibm-ai-platform/micro-g3.3-8b-instruct-1b] FAILED```

Seems flaky since running only CB tests made them all pass 🤔

**Comment by prashantgupta24** [comment]
Too much flakiness going on, going to convert it to a draft for now.

**Comment by prashantgupta24** [comment]
bot:test
